src/mixins/nr-affiliation-mixin.ts

```diff
@@ -1,68 +1,80 @@
-      const nrResponse = await AuthServices.createNRAffiliation(currentOrganizationId, requestBody)
-      if (nrResponse?.status === 200 || nrResponse?.status === 201) {
-        // update the legal api if the status is success
-        const filingBody: BusinessRequest = {
-          filing: {
-            header: {
-              name: 'incorporationApplication',
-              accountId: currentOrganizationId
-            },
-            business: {
-              legalType: (nr.entity_type_cd === 'BC') ? 'BEN' : nr.entity_type_cd
-            },
-            incorporationApplication: {
-              nameRequest: {
-                legalType: (nr.entity_type_cd === 'BC') ? 'BEN' : nr.entity_type_cd,
-                nrNumber: nr.nrNum
-              }
-            }
-          }
+      //
+      // try to create the (temporary) business
+      //
+      const name = this.isBenefitCompany(nr) ? 'incorporationApplication' : 'registration'
+      const legalType = this.isBenefitCompany(nr) ? 'BEN' : nr.entity_type_cd
+      const nrNumber = nr.nrNum
+      const businessRequest: BusinessRequest = {
+        filing: {
+          header: { name, accountId },
+          business: { legalType },
+          registration: this.isFirm(nr) ? { nameRequest: { legalType, nrNumber } } : undefined,
+          incorporationApplication: !this.isFirm(nr) ? { nameRequest: { legalType, nrNumber } } : undefined
         }
-        const filingResponse = await BusinessServices.createBusiness(filingBody)
-        // navigate to My Business Registry dashboard
-        if (filingResponse?.status === 200 || filingResponse?.status === 201) {
-          navigate(
-            `${sessionStorage.getItem('BUSINESSES_URL')}account/${currentOrganizationId}/business`
-          )
-        } else throw new Error('Business creation error: invalid api response ')
-      } else throw new Error('Affiliation error: invalid api response ')
+      }
+
+      const createResponse = await BusinessServices.createBusiness(businessRequest)
+      if (createResponse.status !== CREATED) {
+        // because create failed, delete the existing affiliation to avoid orphan records
+        await AuthServices.removeNrAffiliation(accountId, nrNumber).catch(() => null)
+        throw new Error('Unable to create business')
+      }
+
+      //
+      // navigate to entity dashboard (Filings UI)
+      //
+      const dashboardUrl = sessionStorage.getItem('DASHBOARD_URL')
+      const businessId = createResponse?.data?.filing?.business?.identifier
+      navigate(`${dashboardUrl}${businessId}`)
     } catch (err) {
       console.error('createAffiliation() =', err) // eslint-disable-line no-console
 
       // hide spinner
       this.$root.$emit('showSpinner', false)
 
-      // clear NR Data and show error dialog
+      // clear NR data and show error dialog
       sessionStorage.removeItem('NR_DATA')
       this.setAffiliationErrorModalVisible(true)
```

Comment:
This modal is displayed even when there is a createBusiness() error, so it displays an invalid message.

I didn't fix this here but I didn't break it more, either :)

Maybe this can be fixed in the future.